src/Persistence_matrix/test/pm_matrix_tests.h

```diff
@@ -1443,6 +1443,34 @@ void test_barcode() {
   BOOST_CHECK_EQUAL(std::get<2>(*it), 6);
   ++it;
   BOOST_CHECK(it == bars.end());
+
+  //same test but with std::get
+  bars.clear();
+  for (auto it = barcode.begin(); it != barcode.end(); ++it) {
+    bars.emplace(std::get<2>(*it), std::get<0>(*it), std::get<1>(*it));
+  }
+  it = bars.begin();
+  BOOST_CHECK_EQUAL(std::get<0>(*it), 0);
+  BOOST_CHECK_EQUAL(std::get<1>(*it), 0);
+  BOOST_CHECK_EQUAL(std::get<2>(*it), -1);
```

Comment:
Could you also test that [structured binding](https://en.cppreference.com/w/cpp/language/structured_binding) works?
`auto [ x, y, z ] = *it` and then check the values of x, y and z.

Reply:
I had to move things a bit, but I was able to make it work.